sw/host/opentitanlib/src/io/spi.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+
+/// Represents the SPI transfer mode.
+pub enum TransferMode {
+    Mode0,
+    Mode1,
+    Mode2,
+    Mode3,
+}
+
+/// Represents a SPI transfer.
+pub enum Transfer<'a, 'b> {
+    Read(&'a mut [u8]),
+    Write(&'b [u8]),
+    Both(&'a mut [u8], &'b [u8]),
```

Comment:
But, which is which...? (read v write)

Reply:
Changed the lifetime names to hint which ref is read/write.